packages/flutter_tools/test/integration.shard/swift_package_manager_test.dart

```diff
@@ -29,873 +29,881 @@ void main() {
-            flutterBin,
-            appDirectoryPath,
-            options: <String>[platformName, '--debug', '-v'],
-            expectedLines: SwiftPackageManagerUtils.expectedLines(
-              platform: platformName,
-              appDirectoryPath: appDirectoryPath,
-              cocoaPodsPlugin: integrationTestPlugin,
-            ),
-            unexpectedLines: SwiftPackageManagerUtils.unexpectedLines(
-              platform: platformName,
-              appDirectoryPath: appDirectoryPath,
-              cocoaPodsPlugin: integrationTestPlugin,
-            ),
-          );
-        } finally {
+        addTearDown(() async {
           await SwiftPackageManagerUtils.disableSwiftPackageManager(flutterBin, workingDirectoryPath);
           ErrorHandlingFileSystem.deleteIfExists(
             workingDirectory,
             recursive: true,
           );
-        }
-      }, skip: !platform.isMacOS); // [intended] Swift Package Manager only works on macos.
-    }
+        });
 
-    test('Build $platformName-framework with non-module app uses CocoaPods', () async {
-      final Directory workingDirectory = fileSystem.systemTempDirectory
-          .createTempSync('swift_package_manager_build_framework.');
-      final String workingDirectoryPath = workingDirectory.path;
-      try {
-        // Create and build an app using the Swift Package Manager version of
+        // Create and build an app using the CocoaPods version of
         // integration_test.
-        await SwiftPackageManagerUtils.enableSwiftPackageManager(flutterBin, workingDirectoryPath);
-
+        await SwiftPackageManagerUtils.disableSwiftPackageManager(flutterBin, workingDirectoryPath);
         final String appDirectoryPath = await SwiftPackageManagerUtils.createApp(
           flutterBin,
           workingDirectoryPath,
-          iosLanguage: 'swift',
+          iosLanguage: iosLanguage,
           platform: platformName,
           usesSwiftPackageManager: true,
           options: <String>['--platforms=$platformName'],
         );
         SwiftPackageManagerUtils.addDependency(appDirectoryPath: appDirectoryPath, plugin: integrationTestPlugin);
+        await SwiftPackageManagerUtils.buildApp(
+          flutterBin,
+          appDirectoryPath,
+          options: <String>[platformName, '--debug', '-v'],
+          expectedLines: SwiftPackageManagerUtils.expectedLines(
+            platform: platformName,
+            appDirectoryPath: appDirectoryPath,
+            cocoaPodsPlugin: integrationTestPlugin,
+          ),
+          unexpectedLines: SwiftPackageManagerUtils.unexpectedLines(
+            platform: platformName,
+            appDirectoryPath: appDirectoryPath,
+            cocoaPodsPlugin: integrationTestPlugin,
+          ),
+        );
+        expect(
+          fileSystem
+            .directory(appDirectoryPath)
+            .childDirectory(platformName)
+            .childFile('Podfile')
+            .existsSync(),
+          isTrue,
```

Comment:
I'd go one step further and use `exists` (it's used elsewhere in `package:flutter_tools`):
https://pub.dev/documentation/file_testing/latest/file_testing/exists-constant.html

Reply:
TIL! The whitespace changes in this one makes this hard to review, I'll send a follow-up to the follow-up 😂